internal/types/host/keepalived_host_collector_host.go

```diff
@@ -0,0 +1,270 @@
+package host
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/sirupsen/logrus"
+
+	"github.com/cafebazaar/keepalived-exporter/internal/collector"
+	"github.com/cafebazaar/keepalived-exporter/internal/types/utils"
+)
+
+var sigNumSupportedVersion = version.Must(version.NewVersion("1.3.8"))
+
+// KeepalivedHostCollectorHost implements Collector for when Keepalived and Keepalived Exporter are both on a same host
+type KeepalivedHostCollectorHost struct {
+	pidPath string
+	version *version.Version
+	useJSON bool
+
+	SIGJSON  syscall.Signal
+	SIGDATA  syscall.Signal
+	SIGSTATS syscall.Signal
+}
+
+// NewKeepalivedHostCollectorHost is creating new instance of KeepalivedHostCollectorHost
+func NewKeepalivedHostCollectorHost(useJSON bool, pidPath string) *KeepalivedHostCollectorHost {
+	k := &KeepalivedHostCollectorHost{
+		useJSON: useJSON,
+		pidPath: pidPath,
+	}
+
+	var err error
+	k.version, err = k.getKeepalivedVersion()
+	if err != nil {
+		logrus.WithError(err).Warn("Version detection failed. Assuming it's the latest one.")
+	}
+
+	k.getSignals()
+
+	return k
+}
+
+func (k *KeepalivedHostCollectorHost) getSignals() {
```

Comment:
I think all the collectors should duplicate this function because the `sigNum()` implementation is just different. Am I wrong?

Reply:
You're right. The `sigNum` would be different for each implementation. Although it's possible to prevent duplicating this method, the function is so small that I don't think it's worth the effort. :D

Reply:
Why not move the signal variables to `KeepalivedCollector`?

Reply:
I'd like to think that the different implementations of the `Collector` interface are responsible for communicating with the keepalived in whatever way and keeping track of the signals should be their responsibility, no matter how much is the same between different implementations.  However it could be a utils function.

Reply:
This makes sense to the concept of the function but do you think it's the same for the variables? I mean the variables are the same because Keepalived has fixed signal numbers for its data exporting.

Reply:
Sure. The `collector` package should be only concerned about exporting data. and the types packages should handle all the communicating parts with keepalived.